zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -97,65 +92,164 @@
     @Setter
     private String message;
 
+    @In
+    private RequestService requestServiceImpl;
+
+    @In
+    private ZanataIdentity identity;
+
+    @In
+    private LanguageTeamService languageTeamServiceImpl;
+
     @In(value = ZanataJpaIdentityStore.AUTHENTICATED_USER, required = false)
     private HAccount authenticatedAccount;
 
-    public boolean hasSelectedRole() {
-        return requestAsTranslator || requestAsReviewer || requestAsCoordinator;
+    public String getMyRequestedRole() {
```

Comment:
- [ ] SF: Where is this used? Seems like a better name would be `getMyRequestedRoles()` (plural).